src/main/java/spleetwaise/transaction/model/transaction/Date.java

```diff
@@ -15,9 +16,10 @@
 public class Date {
     public static final String MESSAGE_CONSTRAINTS = "Date should only in the format of DDMMYYYY";
 
-    public static final DateTimeFormatter VALIDATION_FORMATTER = DateTimeFormatter.ofPattern("ddMMyyyy");
+    public static final DateTimeFormatter VALIDATION_FORMATTER = DateTimeFormatter.ofPattern("ddMMuuuu")
```

Comment:
Just want to learn, what is the reason for the change to uuuu?

Reply:
due to the strict validation rules, `yyyy` refers to year-of-era and `uuuu` refers to year. Apparently `2024` is valid for `uuuu` but not `yyyy`

Reply:
I see, could you point us to the documentation?

Reply:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html

<img width="730" alt="Screenshot 2024-11-04 at 4 30 45 PM" src="https://github.com/user-attachments/assets/9b5fc116-a876-4c95-8c5f-e34e5eb88b6f">

Initially though `yyyy` would work but with the strict requirement only `uuuu` works